dashboard/src/components/Config/AppRepoList/AppRepoAddDockerCreds.tsx

```diff
@@ -50,6 +50,12 @@ export function AppRepoAddDockerCreds({
   const handleServerChange = (e: React.ChangeEvent<HTMLInputElement>) => setServer(e.target.value);
   const toggleCredSubForm = () => setShowSecretSubForm(!showSecretSubForm);
 
+  useEffect(() => {
+    if (imagePullSecrets.length && !currentImagePullSecrets.length) {
```

Comment:
This condition appears to only be true if there were no image pull secrets previously? What if a single secret is changed, or a secret is added (not sure if that's supported, but if not, a comment justifying the simpler condition would help), or if a secret is removed? Seems in those cases we wouldn't be setting the current pull secrets?

Reply:
So perhaps we should compare both objects and merge them, instead of replacing it the `currentImagePullSecrets` is empty?

Reply:
no, this is just the `useEffect` of the component, the safe-guard is to avoid re-setting in the component state `currentImagePullSecrets` (and force a re-render) if it's been already set.